support/@trustfractal/data-host/src/popup/services/DataHost.ts

```diff
@@ -0,0 +1,96 @@
+import { Storage, StorageArray } from '@utils/StorageArray';
+
+export class DataHost {
```

Comment:
This is the type of thing we should write in Rust. If we want to include it as TS for now, just to make progress, that's fine. But we should move this into a Rust crate.